darwin/future/data_objects/workflow.py

```diff
@@ -53,8 +53,15 @@ class WFEdge(DefaultDarwin):
 
     id: UUID
     name: str
-    source_stage_id: UUID
-    target_stage_id: UUID
+    source_stage_id: Optional[UUID]
```

Comment:
`WFEdge`s only have both `source_stage_id` and `target_stage_id` if they are in the middle of a workflow.  Those at the ends only have one, so added `Optionals` and a validator requiring one or the other.